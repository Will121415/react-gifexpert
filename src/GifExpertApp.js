import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({defaultCategories = []}) => {

    // const [category, setCategory] = useState(['Naruto']); 
    const [category, setCategory] = useState(defaultCategories);

    return (
        <>
            <h1 className="animate__animated animate__fadeInRightBig">GifExpertApp</h1>
            <AddCategory setCategory={setCategory}/>
            <hr />
            <ol>
                {
                    category.map((category) => 
                        <GifGrid
                            key={category}
                            category={category}
                         />
                    )
                }
            </ol>
        </>
    )
}

export default GifExpertApp;