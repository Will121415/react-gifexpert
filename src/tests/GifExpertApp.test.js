import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp'
import '@testing-library/jest-dom';


describe('Pruebas en el componente <GifExpertApp />', () => { 

    test('Debe de mostrar el componente correctamente', () => {
        const wapper = shallow(<GifExpertApp />);
        expect(wapper).toMatchSnapshot();
    })


    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
        const wapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wapper).toMatchSnapshot();
        expect(wapper.find('GifGrid').length).toBe(categories.length);
    })
})