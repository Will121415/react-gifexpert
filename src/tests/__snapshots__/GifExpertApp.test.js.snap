// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__fadeInRightBig"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__fadeInRightBig"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
